### ISSUE TITLE
inference wrong with custom dataset 

### ISSUE THREAD

**Issue by yi0109-sec** [zylo117/Yet-Another-EfficientDet-Pytorch#732]
I trained with custom datasets 
It seemed to be learning 

But if I inference image to detect object 
Even though there is one object in the image, dozens of boxes are almost randomly found regardless of the object.

there is no box that finds the object properly.

If anyone has experienced or solved this problem, please help me.

**Comment by PerlMonker303** [comment]
Hello. I have a similar use-case as yours, trying to train it on a custom dataset. In my experience, a good practice is to overfit the model on purpose first to see if it suitable for your problem. By overfitting I mean to train and validate the model on a single image (or 5, 10, but start with 1) with a single object (you mentioned this) that is easy to detect.

Intuitively, the more epochs you train, the more the loss should decrease and the more the model overfits the data. In the end (after n epochs, find n) you should see the model detect the object almost perfectly.

If this worked, the next step would be to overfit it on multiple objects and only afterwards try to actually train it on the whole dataset.

Hope this helps. Let me know if you encounter any issues.

Best,
Andrei A.R.
